src/sdk/service/network-validation-service/__tests__/network-validation-service.test.unit.ts

```diff
@@ -0,0 +1,388 @@
+  expect(NetworkValidationService.isValidIPv4Address('74.203.180.179', false)).toBeTruthy();
+});
+
+test('Check if IP Address is valid and not reserved.', () => {
+  expect(NetworkValidationService.isValidIPv4Address('74.203.180.255', false)).toBeFalsy();
+});
+
+test('Check if Netmask is valid.', () => {
+  expect(NetworkValidationService.isValidNetmask('255.255.255.128')).toBeTruthy();
+});
+
+test('Check if Netmask is invalid.', () => {
+  expect(NetworkValidationService.isValidNetmask('255.255.0.128')).toBeFalsy();
+});
+
+test('Check if Netmask is invalid.', () => {
+  expect(NetworkValidationService.isValidNetmask('1.0.1.0')).toBeFalsy();
+});
+
+test('Check if CIDR is valid.', () => {
+  expect(NetworkValidationService.isValidCidr(dummyGateway + '/18')).toBeTruthy();
+});
+
+test('Check if CIDR is invalid.', () => {
+  expect(NetworkValidationService.isValidCidr(' 192.30.250.001')).toBeFalsy();
+});
+
+test('Check if CIDR is invalid.', () => {
+  expect(NetworkValidationService.isValidCidr(' 192.30.250.001/32')).toBeFalsy();
+});
+
+test('Check if CIDR is invalid.', () => {
+  expect(NetworkValidationService.isValidCidr(' 192.30.250.001/0')).toBeFalsy();
+});
+
+test('Check if Subnet CIDR is invalid.', () => {
+  expect(NetworkValidationService.isValidSubnetCidr(dummyGateway + '/18')).toBeFalsy();
+});
+
+test('Check if Subnet CIDR is invalid.', () => {
+  expect(NetworkValidationService.isValidSubnetCidr(dummyGateway + '/0')).toBeFalsy();
+});
+
+test('Check if function gets network descriptor from CIDR.', () => {
+  expect(NetworkValidationService.getIPv4NetworkDescriptorFromCidr(dummyGateway + '/18')).toEqual(
+    {
+      gateway: dummyGateway,
+      netmask: dummyNetmaskOne
+    };
+});
+
+test('Check if function fails to get network descriptor.', () => {
+  expect(NetworkValidationService.getIPv4NetworkDescriptorFromCidr(dummyGateway + '/0')).toBeNull();
+});
+
+test('Check if function gets IPv4AddressRange from String.', () => {
+  expect(NetworkValidationService.getIPv4AddressRangeFromString('1.2.3.4 - 5.6.7.8')).toEqual(
+    {
+      end: '5.6.7.8',
+      start: '1.2.3.4'
+    };
+});
+
+test('Get Subnet CIDR.', () => {
+  expect(NetworkValidationService.getSubnetCidr(dummyNetmaskOne, dummyIpThree)).toEqual(
+      '192.168.0.0/18';
+});
+
+test('Check if function fails to gets Subnet CIDR.', () => {
+  expect(NetworkValidationService.getSubnetCidr(dummyNetmaskThree, '0.0.0.0')).toBeNull();
+});
+
+test('Check if function fails to gets Subnet CIDR.', () => {
+  expect(NetworkValidationService.getSubnetCidr(dummyNetmaskOne, '192.168.0.?')).toBeNull();
+});
+
+test('Check if function fails to get subnet IPv4 Address range.', () => {
+  expect(NetworkValidationService.getSubnetIPv4AddressRange('255.255.192.0', dummyIpThree)).toEqual(
+    {
+      end: '192.168.63.255',
+      start: '192.168.0.0'
+    };
+});
+
+test('Check if function fails to get subnet IPv4 Address range.', () => {
```

Comment:
the description here is identical to the previous one. You should be putting multiple `expect` statements within each test case, where appropriate. And for these types of tests, it's usually appropriate to be testing multiple values in each test case to ensure you're covering all edge cases.